In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns


In [6]:
df = pd.read_csv( "2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv"  )

In [7]:
# Look at the first five rows of the DataFrame
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  
0  2017-03-13 12:36:56  
1  2017-03-03 17:38:52  
2   2017-03-14 4:23:56  
3  2017-03-26 12:43:37  
4   2017-03-01 4:35:11

In [8]:
#check AOV of $3145.13
df['order_amount'].mean()

3145.128

In [9]:
# Look at the statistics of the data
df.describe()

order_id      shop_id      user_id   order_amount  total_items
count  5000.000000  5000.000000  5000.000000    5000.000000   5000.00000
mean   2500.500000    50.078800   849.092400    3145.128000      8.78720
std    1443.520003    29.006118    87.798982   41282.539349    116.32032
min       1.000000     1.000000   607.000000      90.000000      1.00000
25%    1250.750000    24.000000   775.000000     163.000000      1.00000
50%    2500.500000    50.000000   849.000000     284.000000      2.00000
75%    3750.250000    75.000000   925.000000     390.000000      3.00000
max    5000.000000   100.000000   999.000000  704000.000000   2000.00000

The mean is 3145.128000
However, the 50 percentile is only 284.000000,
and the 75 percentile is only 390.000000
Thus there exists outliers problem

In [10]:
# Look at the top 30 highest order amounts and the number of times they appear
df.order_amount.value_counts().sort_index(ascending=False).head(30)

704000    17
154350     1
102900     1
77175      9
51450     16
25725     19
1760       1
1408       2
1086       1
1064       1
1056       3
980        1
965        1
960        2
948        1
935        3
920        1
890        1
885        2
880        3
865        2
845        1
830        3
816        1
815        2
810        1
805        2
804        2
800        3
790        2
Name: order_amount, dtype: int64

The majority of these values are occuring a small number of times, i.e. 1-3 times, and the ones that are occuring more frequently are still not of real significance being that the dataset is 5000 observations, but it is enough that it will throw off the mean (average) value.

In [11]:
# Create an "outlier" column for any order amount that is greater than the 
#   75 percentile value, adding a 1 to the column if it is an outlier and 
#   a 0 to the column if it is not considered an outlier
df['outlier'] = np.where(orders.order_amount > 390, 1, 0)

# Check my work by looking at the first 5 rows of the orders dataset
df.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  outlier  
0  2017-03-13 12:36:56        0  
1  2017-03-03 17:38:52        0  
2   2017-03-14 4:23:56        0  
3  2017-03-26 12:43:37        0  
4   2017-03-01 4:35:11        0

In [13]:
#create a dataset without outliers, then calculate the mean of order_amount
df_no_outliers = df[df['outlier'] == 0]
df_no_outliers['order_amount'].mean()

230.4593658406608

In [14]:
# What is the median amount for the orders
df.order_amount.median()

284.0

In [18]:
print(f'Average order amount without the outliers: ${230.4593658406608:,.2f}\n')
print( f'Median order amount with outliers: ${df.order_amount.median():,.2f}\n' )
print(
    f'Median order amount without outliers: ${df_no_outliers.order_amount.median():,.2f}\n'
    )

Average order amount without the outliers: $230.46

Median order amount with outliers: $284.00

Median order amount without outliers: $224.00

